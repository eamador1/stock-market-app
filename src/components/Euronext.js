import { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import fetchCompanies from '../api/companiesApi';
import { getId, setSearch } from '../redux/companies/companiesSlice';
import logo from '../images/logo.png';
import SvgBack from '../imageComponent/back';
import '../style/Euronext.css';
import SvgForward from '../imageComponent/forward';

const Euronext = () => {
  const {
    isLoading, error, data,
  } = useSelector((state) => state.home);
  const { search } = useSelector((state) => state.home);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCompanies());
    dispatch(setSearch(''));
  }, [dispatch]);

  const handleGetId = (id) => {
    dispatch(getId(id));
  };

  const handleSearch = (e) => {
    dispatch(setSearch(e.target.value));
  };

  return (
    <section className="wrapper">
      <div className="headerContainer">
        <Link to="/" className="exit">
          <SvgBack alt="exit" />
        </Link>
        <h3 className="exchange">EURONEXT</h3>
      </div>
      <div className="search-bar">
        <input className="inputName" type="text" placeholder="Company's Symbol ..." onChange={(e) => handleSearch(e)} />
      </div>
      {isLoading && (
      <div className="loadingImage">
        <img src={logo} className="App-logo" alt="logo" />
        <h2> Loading ...</h2>
      </div>
      )}
      {error && <p className="">{error}</p>}
      <div className="headerCompanies">
        <h2>COMPANIES</h2>
      </div>
      <div className="data">
        {data
            && data
              .filter((list) => {
                const searchLower = search.toLowerCase();
                const listSymbolLower = list.symbol.toLowerCase();
                return searchLower === '' ? true : listSymbolLower.includes(searchLower);
              }).map((item) => (
                <div className="oneCompanyContainer" key={item.id}>
                  <Link className="enter-icon" to="content" onClick={() => handleGetId(item.id)}>
                    <div className="CompanyContainer3" />
                    <div className="right">
                      {' '}
                      <SvgForward />
                      {' '}
                    </div>
                    <div className="left">
                      <p className="symbol">
                        Symbol:
                        { ' ' }
                        {item.symbol}
                      </p>
                      <p className="name">
                        Name:
                        { ' ' }
                        {item.name}
                      </p>
                    </div>

                  </Link>
                </div>
              ))}
      </div>
    </section>
  );
};

export default Euronext;
