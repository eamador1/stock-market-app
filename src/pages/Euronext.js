import React from 'react';

const Euronext = () => (
  <div>
    Euronext Page under Construction
  </div>
);

export default Euronext;
