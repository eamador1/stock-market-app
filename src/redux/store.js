import { configureStore } from '@reduxjs/toolkit';
import companiesReducer from './companies/companiesSlice';

const store = configureStore({
  reducer: {
    home: companiesReducer,
  },
});

export default store;
