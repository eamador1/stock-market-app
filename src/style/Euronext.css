@import url('https://fonts.googleapis.com/css2?family=Proza+Libre:wght@400;500;600;700;800&family=Roboto:ital,wght@0,400;0,500;0,700;1,700&display=swap');

:root {
  background-color: #f36099;
}

*,
body {
  margin: 0;
  padding: 0;
  text-decoration: none;
}

.wrapper {
  padding-bottom: 20vh;
}

.headerContainer {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  padding-top: 2vh;
}

.loadingImage {
  width: 90px;
  display: flex;
  padding-left: 48vw;
  padding-top: 3vh;
}

.exchange {
  color: #ffff;
  font-family: 'Proza Libre', sans-serif;
  font-weight: 700;
  font-size: 30px;
  margin-left: 20vw;
}

.headerCompanies {
  display: flex;
  justify-content: center;
  padding: 3vh 0 3vh 0;
  color: #ffff;
}

.search-bar {
  display: flex;
  justify-content: center;
}

.inputName {
  border: none;
  border-radius: 4px;
  width: 50vw;
  height: 5vh;
  text-indent: 10px;
}

.inputName::placeholder {
  padding-left: 10px;
}

.companyContainer3 {
  display: flex;
}

.oneCompanyContainer {
  padding-left: 5%;
  padding-top: 0;
  margin-top: 0;
  padding-bottom: 1vh;
}

.oneCompanyContainer:nth-child(odd) {
  background-color: #a34368;
}

.oneCompanyContainer:nth-child(even) {
  background-color: #ce5281;
}

.oneCompanyContainer:nth-child(odd):hover {
  background-color: #aa0a48;
}

.oneCompanyContainer:nth-child(even):hover {
  background-color: #ef79a6;
}

.symbol {
  color: #fff;
  width: 60vw;
  font-family: 'Roboto', sans-serif;
}

.left {
  display: flex;
  flex-direction: column;
  width: 70vw;
  margin-top: 0;
  padding-top: 0;
}

.right {
  display: flex;
  justify-content: flex-end;
  width: 90vw;
  margin-top: 0;
  padding-top: 0;
}

.name {
  color: #fff;
  width: 80vw;
  font-family: 'Roboto', sans-serif;
}

@media (min-width: 768px) {
  .inputName {
    width: 20vw;
    height: 4vh;
    text-indent: 10px;
  }

  .exchange {
    padding-left: 22vw;
  }

  .symbol {
    width: 10vw;
  }

  .name {
    width: 20vw;
  }

  .left {
    width: 95vw;
  }

  .right {
    width: 90vw;
  }
}
