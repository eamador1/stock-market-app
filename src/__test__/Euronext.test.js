import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import store from '../redux/store';
import Euronext from '../components/Euronext';
import Detail from '../components/CompanyDetail';

describe('Renders without crashing', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Euronext />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Euronext should render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/euronext" element={<Euronext />} />
            <Route path="/euronext/:content" element={<Detail />} />
          </Routes>
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
