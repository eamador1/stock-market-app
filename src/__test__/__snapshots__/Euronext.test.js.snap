// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing Euronext should render correctly 1`] = `null`;

exports[`Renders without crashing Home should render correctly 1`] = `null`;

exports[`Renders without crashing should render correctly 1`] = `
<section
  className="wrapper"
>
  <div
    className="headerContainer"
  >
    <a
      className="exit"
      href="/"
      onClick={[Function]}
    >
      <svg
        className="w-6 h-6"
        fill="none"
        stroke="#ffffff"
        strokeWidth="1.5"
        viewBox="0 0 24 24"
        width="2em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.75 19.5L8.25 12l7.5-7.5"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </a>
    <h3
      className="exchange"
    >
      EURONEXT
    </h3>
  </div>
  <div
    className="search-bar"
  >
    <input
      className="inputName"
      onChange={[Function]}
      placeholder="Company's Symbol ..."
      type="text"
    />
  </div>
  <div
    className="headerCompanies"
  >
    <h2>
      COMPANIES
    </h2>
  </div>
  <div
    className="data"
  />
</section>
`;
